newIDE/app/src/MainFrame/EditorContainers/EventsEditorContainer.js

```diff
@@ -62,25 +63,27 @@ export class EventsEditorContainer extends React.Component<RenderEditorContainer
     }
 
     return (
-      <EventsSheet
-        ref={editor => (this.editor = editor)}
-        setToolbar={this.props.setToolbar}
-        onOpenLayout={this.props.onOpenLayout}
-        resourceSources={this.props.resourceSources}
-        onChooseResource={this.props.onChooseResource}
-        resourceExternalEditors={this.props.resourceExternalEditors}
-        openInstructionOrExpression={this.props.openInstructionOrExpression}
-        onCreateEventsFunction={this.props.onCreateEventsFunction}
-        unsavedChanges={this.props.unsavedChanges}
-        project={project}
-        scope={{
-          layout,
-        }}
-        globalObjectsContainer={project}
-        objectsContainer={layout}
-        events={layout.getEvents()}
-        onOpenExternalEvents={this.props.onOpenExternalEvents}
-      />
+      <LeaderboardProvider gameId={project.getProjectUuid()}>
```

Comment:
Do we really need the provider to be here, couldn't there be a single LeaderboardProvider at the top of the app? (Providers.js)

Reply:
I don't know if it would conflict with the provider that provides the game dashboard. We wouldn't want the user to select a leaderboard for a different game

Reply:
I think the "bottom most" provider is used when you `useContext`. I agree we don't want to mix leaderboards between games. And that if we put the provider too deep in the tree (like in the field), well you have to reload leaderboards every time a field is rendered.

Reply:
Isn't it a bit strange to place the Provider on top of the App even though you may start the app with no project (it needs a gameId and it seems a bit complex to retrieve the gameId because of the difference browser/local app) ?
I suggest we make it wrap the editorsTabs in MainFrame instead and keep the fetch call in EventsSheet ?